### ISSUE TITLE
baresip should not automatically make call to contact of 3xx response

### ISSUE THREAD

**Issue by juha-h** [baresip/baresip#2597]
When baresip receives 3xx response, it should send UA_EVENT_CALL_REDIRECT to be handled by the application (e.g. menu module).  Making blind transfer without user consultation is dangerous and now there is no possibility to avoid it.

I would be willing to try make the change, but did not understand how this piece of code
```
ua_event(call->ua, UA_EVENT_CALL_BLIND_TRANSFER, call,
		"%d - %s", msg->scode, uri);
```
 in call.c redirect_handler() currently causes the blind transfer.  There is no reference to UA_EVENT_CALL_BLIND_TRANSFER in menu module or anywhere else outside of event.c.

**Comment by juha-h** [comment]
Is the automatic initiation of new call build deep into this piece of code in `re/sipsess/invite_resp_handler()`?
```
	else if (msg->scode < 400) {

		/* Redirect to first Contact */

		if (sess->terminated)
			goto out;

		err = sip_dialog_update(sess->dlg, msg);
		if (err)
			goto out;

		if (sess->redirecth)
			sess->redirecth(msg, sip_dialog_uri(sess->dlg),
				        sess->arg);

		err = invite(sess);
		if (err)
			goto out;

		return;
	}
```
If so, how can it be overridden?

**Comment by cspiel1** [comment]
It's only a name. Should we change it to `UA_EVENT_CALL_REDIRECT`?

Baresip does nothing with this event, except passing to the application. The application/user may decide if the redirect target should be contacted. There is no issue. Should we close it?

**Comment by juha-h** [comment]
Yes, baresip does nothing with the event, but (as I quoted the code above) re re-directs the call without giving any change to baresip app to decide what to do with it.  This is major security issue.

**Comment by cspiel1** [comment]
> baresip should not automatically make call to contact of 3xx response

The title is miss leading. Baresip does not make a call to the 3xx Contact.


> Yes, baresip does nothing with the event, but (as I quoted the code above) re re-directs the call without giving any change to baresip app to decide what to do with it. This is major security issue.

The user of `re` may set the `redirecth`. There is no security problem.

**Comment by cspiel1** [comment]
Sorry, I understood now. The issue is the `invite()` in `re/src/sipsess/connect.c:169`.

**Comment by alfredh** [comment]
There is a difference between SIP redirect and SIP transfer.

Dealing with 3xx response is not related to "Blind Transfer".


The HTTP protocol also supports 3xx redirect. When your web-browser receiver 3xx it automatically
goes to the new URL, it does not prompt the user.

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> There is a difference between SIP redirect and SIP transfer.
> 
> Dealing with 3xx response is not related to "Blind Transfer".
> 
> 
> The HTTP protocol also supports 3xx redirect. When your web-browser
> receiver 3xx it automatically goes to the new URL, it does not prompt
> the user.

Yes, redirect is not the same transfer.  Usually transfer is made after
call is established.  I haven't tested if baresip accepts REFER as
out-of-dialog request.  If (like always with REFER), confirmation should
be asked from the user before the INVITE is sent.

Redirect is response to INVITE and again confirmation from user should
be asked.  You don't need to pay when you are redirected to another web
page but you may need to pay alot when you are redirected to another
phone number.

**Comment by alfredh** [comment]
To me it looks like the logical choice here is to add a new config option:

```
sip_follow_redirect    yes|no
```

Then users can choose their wanted behaviour.
If other SIP softphones have this setting, we can also copy the name etc..
